catalog/app/components/TalkToUs/TalkToUs.js

```diff
@@ -36,42 +36,50 @@ function getCalendlyPromise() {
   return window[CALENDLY_PROMISE]
 }
 
-// const getCalendlyEvent = (e) =>
-//   e.data.event &&
-//   e.data.event.startsWith('calendly.') &&
-//   e.data.event.substring('calendly.'.length)
+function useSingletonListener() {
+  const ref = React.useRef(null)
+  return (eventName, callback) => {
+    if (ref.current) {
+      window.removeEventListener(eventName, ref.current)
+    }
+    ref.current = callback
+    window.addEventListener(eventName, callback)
+  }
+}
+
+const getCalendlyEvent = (e) =>
+  e?.data?.event?.startsWith('calendly.') && e?.data?.event?.substring('calendly.'.length)
 
 export function TalkToUsProvider({ children }) {
   const cfg = useConfig()
   const t = useTracker()
 
   const calendlyP = cfg.calendlyLink ? getCalendlyPromise() : null
 
+  const listen = useSingletonListener()
+
   const showPopup = React.useCallback(
     (extra) => {
       t.track('WEB', { type: 'meeting', action: 'popup', ...extra })
 
+      const handleCalendlyEvent = (e) => {
+        if (getCalendlyEvent(e) === 'event_scheduled') {
+          t.track('WEB', { type: 'meeting', action: 'scheduled', ...extra })
+        }
+      }
+
       if (!cfg.calendlyLink) {
         // eslint-disable-next-line no-console
         console.warn('Unable to open Calendly popup: missing Config.calendlyLink', extra)
         return
       }
 
-      // function handleCalendlyEvent(e) {
-      //   if (getCalendlyEvent(e) === 'event_scheduled') {
-      //     t.track('WEB', { type: 'meeting', action: 'scheduled', ...extra })
-      //   }
-      // }
-
       calendlyP.then((C) => {
-        // const handleClose = () => {
-        //   window.removeEventListener('message', handleCalendlyEvent)
-        // }
-        // window.addEventListener('message', handleCalendlyEvent)
+        listen('message', handleCalendlyEvent)
         C.initPopupWidget({ url: cfg.calendlyLink })
       })
     },
-    [t, cfg.calendlyLink, calendlyP],
+    [t, cfg.calendlyLink, calendlyP, listen],
```

Comment:
since you're using `listen` in the dependency list, it should have a stable identity (`useSingletonListener`'s return value should be memoized via `useCallback` or similar), otherwise memoizing `showPopup` doesn't make sense